src/utils/__tests__/datetime.test.ts

```diff
@@ -153,13 +157,13 @@ describe('getOrderOfEvent', () => {
   test('3 events middle', () => {
     const event = events[6]
     const index = utils.getOrderOfEvent(event, events)
-    expect(index).toEqual(1)
+    expect(index).toEqual(2)
   })
 
   test('3 events end', () => {
     const event = events[5]
     const index = utils.getOrderOfEvent(event, events)
-    expect(index).toEqual(2)
+    expect(index).toEqual(1)
```

Comment:
Btw as you can see i have adjusted the tests here to satisfy new changes.

Although as you will correctly assume that the changes that i did possibly broke the functionality.

To be able to understand why i passed those test events in rndemo project to be able to visualize it. (i adjusted year from 2021 to 2024 so i dont have to scroll forever to get there)

Screenshot is before changes
![image](https://github.com/user-attachments/assets/6ad3b9bb-ad81-4682-bd1d-58df62fc63f2)

Screenshot after changes
![Screenshot_1733403264](https://github.com/user-attachments/assets/c3767fb5-e56b-40ce-bf31-dad26f94a21c)

As we can see we have three overlapping events which need to be sorted based on chronological order.

Let's analyse the events from testEvents:
"Repair my car" (2024-09-18T05:45 to 2024-09-18T11:30)
"Meet Realtor" (2024-09-18T06:25 to 2024-09-18T07:55)
"Laundry" (2024-09-18T06:25 to 2024-09-18T09:00)
These three events overlap with each other.

Sorting them in order of start time:

"Repair my car" (Start: 05:45, End: 11:30)
"Meet Realtor" (Start: 06:25, End: 07:55)
"Laundry" (Start: 06:25, End: 09:00)
Thus, the sorted order of overlapping events is:

Repair my car (2024-09-18T05:45 to 2024-09-18T11:30)
Meet Realtor (2024-09-18T06:25 to 2024-09-18T07:55)
Laundry (2024-09-18T06:25 to 2024-09-18T09:00)

Default Sorting Criteria for Overlapping Events:
Primary Criterion: Start Time
Events are primarily sorted by their start time (earlier start time comes first).

Secondary Criterion: End Time (Tie-Breaker)
When two events have the same start time, the event with the earlier end time takes precedence. This ensures shorter events come first.

Applying this to "Meet Realtor" and "Laundry":
Start Times: Both start at 06:25, so we move to the secondary criterion.
End Times:
"Meet Realtor" ends at 07:55.
"Laundry" ends at 09:00.
Since "Meet Realtor" ends earlier, it is sorted first.